components/cloud-event-gateway-proxy/pkg/signals/signals.go

```diff
@@ -0,0 +1,71 @@
+package signals
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+var (
+	onlyOneSignalHandler = make(chan struct{})
+	shutdownSignals      = []os.Signal{os.Interrupt, syscall.SIGTERM}
+)
+
+// SetupSignalHandler registered for SIGTERM and SIGINT. A stop channel is returned
+// which is closed on one of these signals. If a second signal is caught, the program
+// is terminated with exit code 1.
+func SetupSignalHandler() (stopCh <-chan struct{}) {
+	close(onlyOneSignalHandler) // panics when called twice
+
+	stop := make(chan struct{})
+	osSignal := make(chan os.Signal, 2)
+	signal.Notify(osSignal, shutdownSignals...)
+	go func() {
+		<-osSignal
+		close(stop)
+		<-osSignal
+		os.Exit(1) // second signal. Exit directly.
+	}()
+
+	return stop
+}
+
+// NewContext creates a new context with SetupSignalHandler()
+// as our Done() channel.
+func NewContext() context.Context {
+	return &signalContext{stopCh: SetupSignalHandler()}
+}
+
+type signalContext struct {
```

Comment:
To improve the code readability, it's better to move the definition of "signalContext" before the first usage of it, in this case before "NewContext()".

Reply:
Done.